crates/uv-publish/src/lib.rs

```diff
@@ -239,34 +242,73 @@ impl PublishSendError {
     }
 }
 
-/// Collect the source distributions and wheels for publishing.
-///
-/// Returns the path, the raw filename and the parsed filename. The raw filename is a fixup for
-/// <https://github.com/astral-sh/uv/issues/8030> caused by
-/// <https://github.com/pypa/setuptools/issues/3777> in combination with
-/// <https://github.com/pypi/warehouse/blob/50a58f3081e693a3772c0283050a275e350004bf/warehouse/forklift/legacy.py#L1133-L1155>
+/// Represents a single "to-be-uploaded" distribution, alone with zero
+/// or more attestations that will be uploaded alongside it.
+#[derive(Debug)]
+pub struct UploadDistribution {
+    /// The path to the main distribution file to upload.
+    pub file: PathBuf,
+    /// The raw filename of the main distribution file.
+    pub raw_filename: String,
+    /// The parsed filename of the main distribution file.
+    pub filename: DistFilename,
+    /// Zero or more paths to PEP 740 attestations for the distribution.
+    pub attestations: Vec<PathBuf>,
+}
+
+/// Given a list of paths (which may contain globs), unroll them into
+/// a flat, unique list of files. Files are returned in a stable
+/// but unspecified order.
 #[allow(clippy::result_large_err)]
```

Comment:
This slipped through last time but we should box large error variants: https://gist.github.com/konstin/c54456482a14c7528829fdf4567040dd